runtime/src/accounts_db.rs

```diff
@@ -4520,18 +4511,13 @@ impl AccountsDb {
             }
             rewrite_elapsed.stop();
 
-            let keep_store_ids = if slot == current_ancient.slot() {
-                // this slot became an ancient append vec, so we want to keep the append vec storage for the slot
-                // and remove the old, normal append vec
-                vec![current_ancient.append_vec_id()]
```

Comment:
Is this where the assert was happening? I'm struggling to see how it would happen here given the call to `create_if_necessary` above.

Reply:
https://github.com/solana-labs/solana/blob/ee0320c9f303f8ce40d3d06c41ab5174ab77bf3a/runtime/src/accounts_db.rs#L4549-L4557

We'd fail this assert because we were throwing away the ancient append vec instead of the original append vec.

Reply:
Awesome, thanks for the clarification!